[ANALYZER-3433]: Drill through does not recognize level's nameColumn …

TatsianaKasiankova opened the pull request:
…attribute

Added the columnName to the drillthrough SQL in extended context.

wingman-pentaho commented:
### Build Completed

:x: This pull request has errors. They will need to be addressed before it can be accepted. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
mvn -B -fn -DrunITs=true -f pom.xml clean install && mvn -B -f pom.xml site
```

#### Cleanup Commands
```sh
mvn -B -f pom.xml build-helper:remove-project-artifact
```

#### Changed files

```console
  mondrian/src/it/java/mondrian/test/DrillThroughTest.java
  mondrian/src/main/java/mondrian/rolap/RolapAggregationManager.java
```

#### Integration test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### mondrian.olap.ParameterImpl.new ParameterSlot() {...}
 * Complexity Change:  + 5.26%
 * Instruction Change:  + 8.43%
 * Line Change:  + 17.65%
 * Method Change:  + 9.09%

###### mondrian.rolap.agg.SegmentCacheManager.Actor
 * Instruction Change: **-4.41%**:small_red_triangle_down:

###### mondrian.rolap.cache.SegmentCacheIndexImpl
 * Branch Change:  + .47%
 * Complexity Change:  + .73%
 * Instruction Change:  + .81%
 * Line Change:  + .26%

###### mondrian.rolap.RolapAggregationManager
 * Branch Change:  + .93%
 * Complexity Change:  + .76%
 * Instruction Change:  + 1.22%
 * Line Change:  + 1.12%

###### mondrian.rolap.RolapStatisticsCache
 * Branch Change:  + 10.71%
 * Complexity Change:  + 14.29%
 * Instruction Change:  + .85%
 * Line Change:  + 3.75%

###### mondrian.server.Execution
 * Instruction Change: **-.76%**:small_red_triangle_down:

###### mondrian.spi.impl.JdbcStatisticsProvider
 * Branch Change:  + 5.00%
 * Complexity Change:  + 13.33%
 * Instruction Change:  + 7.11%
 * Line Change:  + 4.26%
 * Method Change:  + 20.00%

###### mondrian.util.CacheMap
 * Branch Change: **-7.14%**:small_red_triangle_down:
 * Complexity Change: **-4.35%**:small_red_triangle_down:
 * Instruction Change: **-2.81%**:small_red_triangle_down:
 * Line Change: **-3.77%**:small_red_triangle_down:

###### mondrian.util.FilteredIterableList
 * Branch Change: **-5.26%**:small_red_triangle_down:
 * Complexity Change: **-3.57%**:small_red_triangle_down:
 * Instruction Change: **-.75%**:small_red_triangle_down:

#### License header violations

> :heavy_exclamation_mark: Copyright year is not the current year on mondrian/src/main/java/mondrian/rolap/RolapAggregationManager.java. Found 2016. Was expecting 2017

wingman-pentaho commented:
### Build Completed

:fire: This pull request has some issues. It would be preferable to fix them in order for it to be just perfect. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
mvn -B -fn -DrunITs=true -f pom.xml clean install && mvn -B -f pom.xml site
```

#### Cleanup Commands
```sh
mvn -B -f pom.xml build-helper:remove-project-artifact
```

#### Changed files

```console
  mondrian/src/it/java/mondrian/test/DrillThroughTest.java
  mondrian/src/main/java/mondrian/rolap/RolapAggregationManager.java
```

#### Integration test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### mondrian.olap.Util.Flat3List
 * Branch Change:  + 2.63%
 * Complexity Change:  + 3.23%

###### mondrian.rolap.agg.SegmentCacheManager.Actor
 * Instruction Change: **-1.47%**:small_red_triangle_down:
 * Line Change: **-3.45%**:small_red_triangle_down:

###### mondrian.rolap.cache.SegmentCacheIndexImpl
 * Branch Change: **-.47%**:small_red_triangle_down:
 * Complexity Change: **-.73%**:small_red_triangle_down:
 * Instruction Change: **-.81%**:small_red_triangle_down:
 * Line Change: **-.26%**:small_red_triangle_down:

###### mondrian.rolap.RolapAggregationManager
 * Branch Change:  + .93%
 * Complexity Change:  + .76%
 * Instruction Change:  + 1.22%
 * Line Change:  + 1.12%

###### mondrian.server.FileRepository
 * Instruction Change: **-1.00%**:small_red_triangle_down:
 * Line Change: **-2.19%**:small_red_triangle_down:

###### mondrian.server.MondrianServerRegistry
 * Branch Change: **-6.25%**:small_red_triangle_down:
 * Complexity Change: **-6.67%**:small_red_triangle_down:
 * Instruction Change: **-7.19%**:small_red_triangle_down:
 * Line Change: **-2.08%**:small_red_triangle_down:

wingman-pentaho commented:
### Build Completed

:x: This pull request has errors. They will need to be addressed before it can be accepted. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
mvn -B -fn -DrunITs=true -f pom.xml clean install && mvn -B -f pom.xml site
```

#### Cleanup Commands
```sh
mvn -B -f pom.xml build-helper:remove-project-artifact
```

#### Changed files

```console
  mondrian/src/it/java/mondrian/test/DrillThroughTest.java
  mondrian/src/main/java/mondrian/rolap/RolapAggregationManager.java
```

#### Integration test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### mondrian.olap.Util.Flat3List
 * Branch Change: **-2.63%**:small_red_triangle_down:
 * Complexity Change: **-3.23%**:small_red_triangle_down:

###### mondrian.rolap.agg.SegmentCacheManager.AbortException
 * Complexity Change:  + 100.00%
 * Instruction Change:  + 100.00%
 * Line Change:  + 100.00%
 * Method Change:  + 100.00%

###### mondrian.rolap.agg.SegmentCacheManager.Actor
 * Branch Change:  + 25.00%
 * Complexity Change:  + 12.50%
 * Instruction Change:  + 15.69%
 * Line Change:  + 12.07%

###### mondrian.rolap.agg.SegmentLoader
 * Branch Change:  + 1.98%
 * Complexity Change:  + 2.38%
 * Instruction Change:  + 1.28%
 * Line Change:  + .88%

###### mondrian.rolap.agg.SegmentLoader.2.new SegmentCacheManager.Command() {...}
 * Branch Change:  + 33.33%
 * Complexity Change:  + 33.33%
 * Instruction Change:  + 6.78%
 * Line Change:  + 6.67%

###### mondrian.rolap.BatchLoader
 * Branch Change:  + 1.00%
 * Complexity Change:  + 1.61%
 * Instruction Change:  + 1.30%
 * Line Change:  + .94%

###### mondrian.rolap.BatchLoader.BatchComparator
 * Branch Change: **-6.25%**:small_red_triangle_down:
 * Complexity Change: **-8.33%**:small_red_triangle_down:
 * Instruction Change: **-5.41%**:small_red_triangle_down:
 * Line Change: **-3.70%**:small_red_triangle_down:

###### mondrian.rolap.FastBatchingCellReader
 * Branch Change: **-1.96%**:small_red_triangle_down:
 * Complexity Change: **-3.08%**:small_red_triangle_down:
 * Instruction Change: **-.42%**:small_red_triangle_down:
 * Line Change: **-1.18%**:small_red_triangle_down:

###### mondrian.rolap.RolapAggregationManager
 * Branch Change:  + .93%
 * Complexity Change:  + .76%
 * Instruction Change:  + 1.22%
 * Line Change:  + 1.12%

###### mondrian.server.Execution
 * Instruction Change:  + .76%

###### mondrian.server.MonitorImpl.Handler
 * Branch Change:  + 2.27%
 * Complexity Change:  + 1.85%
 * Instruction Change:  + .42%
 * Line Change:  + .50%

###### mondrian.util.CacheMap
 * Branch Change:  + 7.14%
 * Complexity Change:  + 4.35%
 * Instruction Change:  + 2.81%
 * Line Change:  + 3.77%

###### mondrian.util.FilteredIterableList
 * Branch Change:  + 5.26%
 * Complexity Change:  + 3.57%
 * Instruction Change:  + .75%

#### License header violations

> :heavy_exclamation_mark: Copyright year is not the current year on mondrian/src/main/java/mondrian/rolap/RolapAggregationManager.java. Found 2016. Was expecting 2017

wingman-pentaho commented:
### Build Completed

:fire: This pull request has some issues. It would be preferable to fix them in order for it to be just perfect. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
mvn -B -fn -DrunITs=true -f pom.xml clean install && mvn -B -f pom.xml site
```

#### Cleanup Commands
```sh
mvn -B -f pom.xml build-helper:remove-project-artifact
```

#### Changed files

```console
  mondrian/src/it/java/mondrian/test/DrillThroughTest.java
  mondrian/src/main/java/mondrian/rolap/RolapAggregationManager.java
```

#### Integration test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### mondrian.olap.ParameterImpl.new ParameterSlot() {...}
 * Complexity Change:  + 5.26%
 * Instruction Change:  + 8.43%
 * Line Change:  + 17.65%
 * Method Change:  + 9.09%

###### mondrian.olap.Util.Flat3List
 * Branch Change: **-2.63%**:small_red_triangle_down:
 * Complexity Change: **-3.23%**:small_red_triangle_down:

###### mondrian.rolap.agg.AggregationKey
 * Branch Change: **-3.57%**:small_red_triangle_down:
 * Complexity Change: **-4.17%**:small_red_triangle_down:
 * Instruction Change: **-1.15%**:small_red_triangle_down:
 * Line Change: **-2.38%**:small_red_triangle_down:

###### mondrian.rolap.agg.SegmentCacheManager.AbortException
 * Complexity Change:  + 100.00%
 * Instruction Change:  + 100.00%
 * Line Change:  + 100.00%
 * Method Change:  + 100.00%

###### mondrian.rolap.agg.SegmentCacheManager.Actor
 * Branch Change:  + 25.00%
 * Complexity Change:  + 12.50%
 * Instruction Change:  + 8.33%
 * Line Change:  + 5.17%

###### mondrian.rolap.agg.SegmentLoader
 * Branch Change:  + 1.98%
 * Complexity Change:  + 2.38%
 * Instruction Change:  + 1.28%
 * Line Change:  + .88%

###### mondrian.rolap.agg.SegmentLoader.2.new SegmentCacheManager.Command() {...}
 * Branch Change:  + 33.33%
 * Complexity Change:  + 33.33%
 * Instruction Change:  + 6.78%
 * Line Change:  + 6.67%

###### mondrian.rolap.BatchLoader
 * Branch Change:  + 1.00%
 * Complexity Change:  + 1.61%
 * Instruction Change:  + 1.30%
 * Line Change:  + .94%

###### mondrian.rolap.cache.SegmentCacheIndexImpl
 * Branch Change: **-.47%**:small_red_triangle_down:
 * Complexity Change: **-.73%**:small_red_triangle_down:
 * Instruction Change: **-.81%**:small_red_triangle_down:
 * Line Change: **-.26%**:small_red_triangle_down:

###### mondrian.rolap.FastBatchingCellReader
 * Branch Change: **-1.96%**:small_red_triangle_down:
 * Complexity Change: **-3.08%**:small_red_triangle_down:
 * Instruction Change: **-.42%**:small_red_triangle_down:
 * Line Change: **-1.18%**:small_red_triangle_down:

###### mondrian.rolap.RolapAggregationManager
 * Branch Change:  + .93%
 * Complexity Change:  + .76%
 * Instruction Change:  + 1.22%
 * Line Change:  + 1.12%

###### mondrian.util.CacheMap
 * Branch Change: **-7.14%**:small_red_triangle_down:
 * Complexity Change: **-4.35%**:small_red_triangle_down:
 * Instruction Change: **-2.81%**:small_red_triangle_down:
 * Line Change: **-3.77%**:small_red_triangle_down:

###### mondrian.util.FilteredIterableList
 * Branch Change: **-5.26%**:small_red_triangle_down:
 * Complexity Change: **-3.57%**:small_red_triangle_down:
 * Instruction Change: **-.75%**:small_red_triangle_down:

TatsianaKasiankova commented:
@mkambol @lucboudreau please review.

TatsianaKasiankova commented:
@lucboudreau , @mkambol 
Hi, could you please review. I've implemented fix as mentioned in the letter: `ANALYZER-3433: Question about fix` and added unit tests.

mchen-len-son commented:
@lucboudreau , @mkambol , @kurtwalker 
can you please take a look?
thanks.
